react-components/src/architecture/concrete/clipping/commands/NextClippingCommand.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { RenderTargetCommand } from '../../../base/commands/RenderTargetCommand';
+import { type DomainObject } from '../../../base/domainObjects/DomainObject';
+import { type TranslateKey } from '../../../base/utilities/TranslateKey';
+import { CropBoxDomainObject } from '../CropBoxDomainObject';
+import { SliceDomainObject } from '../SliceDomainObject';
+import { ApplyClipCommand } from './ApplyClipCommand';
+
+export class NextClippingCommand extends RenderTargetCommand {
+  // ==================================================
+  // OVERRIDES
+  // ==================================================
+
+  public override get tooltip(): TranslateKey {
+    return {
+      key: 'CLIP_NEXT',
+      fallback: 'Set the next crop box or slicing plane as global clipping'
+    };
+  }
+
+  public override get icon(): string {
+    return 'ArrowRight';
+  }
+
+  public override get isEnabled(): boolean {
+    if (!this.renderTarget.isGlobalClippingActive) {
+      return false;
+    }
+    // Require at least two crop boxes or one crop box and one slice
+    let count = 0;
+    for (const domainObject of this.rootDomainObject.getDescendants()) {
+      if (domainObject instanceof CropBoxDomainObject) {
+        count++;
+      }
+    }
+    if (this.rootDomainObject.getDescendantByType(SliceDomainObject) !== undefined) {
+      count++;
+    }
+    return count >= 2;
+  }
+
+  protected override invokeCore(): boolean {
```

Comment:
There are lots going on in this function, would be nice to separate it with helper functions like
```
getCropBoxesAndSlices()
getSelectedIndex()
and others()
```

Reply:
This is now fixed in 10e758685d41e3b28aef8d9e9745f874bd90ae40